homeassistant/components/permobil/sensor.py

```diff
@@ -0,0 +1,361 @@
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Create sensors from a config entry created in the integrations UI."""
+    config = hass.data[DOMAIN][config_entry.entry_id]
+    if config_entry.options:
+        config.update(config_entry.options)
+    session = hass.helpers.aiohttp_client.async_get_clientsession()
+    # create the API object from the config
+    p_api = MyPermobil(
+        application=APPLICATION,
+        session=session,
+        email=config.get(CONF_EMAIL),
+        region=config.get(CONF_REGION),
+        code=config.get(CONF_CODE),
+        token=config.get(CONF_TOKEN),
+        expiration_date=config.get(CONF_TTL),
+    )
+
+    p_api.self_authenticate()
+
+    p_api_unit: str = KM
+    try:
+        # find out what unit of distance the user has set
+        p_api_unit = str(await p_api.request_item(BATTERY_DISTANCE_UNIT))
+    except (MyPermobilClientException, MyPermobilAPIException) as err:
+        _LOGGER.error("Permobil: %s", err)
+
+    if p_api_unit not in [KM, MILES]:
+        _LOGGER.error("Unknown unit of distance: %s", p_api_unit)
+
+    # translate the Permobils unit of distance to a Home Assistant unit of distance
+    # default to kilometers if the unit is unknown
+    ha_unit = UnitOfLength.MILES if p_api_unit == MILES else UnitOfLength.KILOMETERS
+    user_specific_sensors = [
+        PermobilDistanceLeftSensor(p_api, unit=ha_unit),
+        PermobilMaxDistanceLeftSensor(p_api, unit=ha_unit),
+        PermobilUsageDistanceSensor(p_api, unit=ha_unit),
+        PermobilRecordDistanceSensor(p_api, unit=ha_unit),
+    ]
+
+    # create the sensors that are not user specific
+    non_specific_sensors = [
+        PermobilStateOfChargeSensor(p_api),
+        PermobilStateOfHealthSensor(p_api),
+        PermobilChargingSensor(p_api),
+        PermobilChargeTimeLeftSensor(p_api),
+        PermobilIndoorDriveTimeSensor(p_api),
+        PermobilMaxWattHoursSensor(p_api),
+        PermobilWattHoursLeftSensor(p_api),
+        PermobilUsageAdjustmentsSensor(p_api),
+        PermobilRecordAdjustmentsSensor(p_api),
+    ]
+
+    sensors = non_specific_sensors + user_specific_sensors
+    async_add_entities(sensors, update_before_add=True)
+
+
+class PermobilGenericSensor(SensorEntity):
+    """Representation of a Sensor.
+
+    This implements the common functions of all sensors.
+    """
+
+    _attr_name = "Generic Sensor"
+
+    def __init__(self, permobil: MyPermobil, item: str, unit: str = "") -> None:
+        """Initialize the sensor.
+
+        item: The item to request from the API.
+        unit: The unit of measurement. (optional)
+        """
+        super().__init__()
+        self._permobil = permobil
+        self._item = item
+        if unit:
+            self._attr_native_unit_of_measurement = unit
+
+    async def async_update(self) -> None:
+        """Get battery percentage."""
+        try:
+            self._attr_native_value = await self._permobil.request_item(self._item)
```

Comment:
Is each sensor doing requests on its own? Shouldn't it just do a single request to get all sensors in a single go?

Considering the number of sensors this integration provides, this seems like it will be doing an awful amount of requests.

Reply:
So why not use an DataUpdateCoordinator instead of making each entity responsible for handling it on its own?

Reply:
I think it is fine to make three requests (if they can be updated at the same time).